src/key_io.cpp

```diff
@@ -68,11 +68,18 @@ class DestinationEncoder : public boost::static_visitor<std::string>
     std::string operator()(const CNoDestination& no) const { return {}; }
 };
 
-CTxDestination DecodeDestination(const std::string& str, const CChainParams& params)
+CTxDestination DecodeDestination(const std::string& str, const CChainParams& params, ColorIdentifier* colorId = nullptr)
 {
     std::vector<unsigned char> data;
     uint160 hash;
     if (DecodeBase58Check(str, data)) {
+
+        CScript scriptPubKey(data.begin(), data.end());
+        if(scriptPubKey.IsColoredScript()) {
+            ColorIdentifier cid(GetColorIdFromScript(scriptPubKey));
+            colorId = &cid;
```

Comment:
this is not safe. the pointer is uninitialised and we are assigning the address of a stack variable. make it a reference,
```ColorIdentifier& colorId = ColorIdentifier()```

then by default its type will be TokenType::NONE